homeassistant/components/bluetooth/wrappers.py

```diff
@@ -188,13 +188,10 @@ async def connect(self, **kwargs: Any) -> bool:
             or not self._async_get_backend_for_ble_device(self.__ble_device)
         ):
             assert models.MANAGER is not None
-            wrapped_backend = (
-                self._async_get_backend() or self._async_get_fallback_backend()
-            )
-            self.__ble_device = (
-                await freshen_ble_device(wrapped_backend.device)
-                or wrapped_backend.device
-            )
+            (
+                wrapped_backend,
+                self.__ble_device,
```

Comment:
This should look every time since it will get stuck on a proxy with a free slot if the one with a better rssi is available once the better rssi runs out of slots until the connect raises